### ISSUE TITLE
Multiple 'update' actions work on initial copy

### ISSUE THREAD

**Issue by ondrap** [paf31/purescript-thermite#61]
I am quite new to thermite, so pardon me if this is easy - but I have tried to run multiple `update` in PerformAction and it seems to me it operates on the initial data, so earlier updates are lost. I just copied this example: https://github.com/nwolverson/purescript-thermite-example , changed the `performAction` to

```
performAction Increment _ _ update = do
    update $ \state -> state { counter = state.counter + 1 }
    update $ \state -> state { counter = state.counter + 3 }
```
And it is incrementing by 3, not by 4. Is this correct behaviour? Reading the source code somehow makes me think it shouldn't ignore the first update, but I don't have any experience with react.

**Comment by paf31** [comment]
This definitely doens't seem right, although it seems partially related to https://github.com/paf31/purescript-thermite/pull/55.

I'm planning to revisit the way async state updates work after psc 0.9.1 is released tomorrow. Chances are the new solution will be based on `purescript-coroutines` somehow.

**Comment by paf31** [comment]
I think `writeStateWithCallback` will fix this.

**Comment by paf31** [comment]
This should be fixed in 1.0.
